src/malebolgia.nim

```diff
@@ -96,63 +87,71 @@ type
   FixedChan = object ## channel of a fixed size
     spaceAvailable, dataAvailable: Cond
     L: Lock
-    head, tail, count: int
     data: array[FixedChanSize, PoolTask]
+    lock: array[FixedChanSize, Lock]
+    todo: array[FixedChanSize, bool]
 
 var
   thr: array[ThreadPoolSize-1, Thread[void]] # -1 because the main thread counts too
   chan: FixedChan
   globalStopToken: Atomic[bool]
-  busyThreads: Atomic[int]
 
 proc send(item: sink PoolTask) =
   # see deques.addLast:
-  acquire(chan.L)
-  while chan.count >= FixedChanSize:
-    wait(chan.spaceAvailable, chan.L)
-  if chan.count < FixedChanSize:
-    inc chan.count
-    chan.data[chan.tail] = item
-    chan.tail = (chan.tail + 1) and FixedChanMask
+  while true:
+    for i in 0..high(chan.lock):
+      if not tryAcquire(chan.lock[i]):
+        continue
+      if chan.todo[i]:
+        chan.lock[i].release
+        continue
```

Comment:
@Araq 
> Tbh I have no idea what this array of locks tries to accomplish. The standard solution to speed things up further is "work stealing". I implemented a variation of it in `malebolgia_push` but it was not faster on my benchmarks and more complex.

Instead of make everyone waits for `chan.L`, every thread tries to work with whatever is available in `chan.lock`, only who has the `chan.lock[i]` lock, should change `chan.data[i]`,

Complexity: now only two lines in the code locks chan, `chan.count`, `chan.head`, `chan.tail` aren't required anymore.

Speed: the only thing that prevents it to be faster than PR19 is line 145 that also removes requirement for 240, but may cause 100% of cpu usage with unused threads. [data](https://gist.github.com/hugosenari/8211707277abd9e02381112e8b088c26)
![image](https://github.com/Araq/malebolgia/assets/863299/9b5e8950-f780-40d2-9557-3424226348e9)
![image](https://github.com/Araq/malebolgia/assets/863299/c99f2722-366d-4c0f-ba78-6baa35cd2fe7)
![image](https://github.com/Araq/malebolgia/assets/863299/739106d5-6d15-4aef-b4e9-a7b9a64b413b)